os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -104,29 +105,24 @@ binmgr_kinfo_t *binary_manager_get_kdata(void)
  *	 This function registers user binaries.
  *
  ****************************************************************************/
-int binary_manager_register_ubin(char *name)
+int binary_manager_register_ubin(char *name, char *version, uint8_t priority)
 {
-	int bin_idx;
-
 	if (name == NULL || g_bin_count >= USER_BIN_COUNT) {
 		bmdbg("ERROR: Invalid parameter\n");
 		return ERROR;
 	}
 
-	for (bin_idx = 1; bin_idx <= g_bin_count; bin_idx++) {
-		/* Already Registered */
-		if (!strncmp(BIN_NAME(bin_idx), name, strlen(name) + 1)) {
-			bmdbg("Already registered for binary %s\n", BIN_NAME(bin_idx));
-			return ERROR;
-		}
-	}
-
 	/* If partition is not registered, Register it as a new user partition */
 	g_bin_count++;
 	BIN_ID(g_bin_count) = -1;
+	BIN_FILECNT(g_bin_count) = 1;
 	BIN_RTLIST(g_bin_count) = NULL;
 	BIN_NRTLIST(g_bin_count) = NULL;
+	BIN_PRIORITY(g_bin_count) = priority;
 	BIN_STATE(g_bin_count) = BINARY_INACTIVE;
+	if (version != NULL) {
```

Comment:
Most of the other params are checked in the first of the function. I think we can check version also in the first, if there can be a problem without version. (I think if there is no version, it is not a valid one.)

Reply:
It is for new binary to be downloaded. We cannot know the version at this time.

Reply:
You can use `version` even for new binary.